File: output/GPT4o-Mini/xCodeEval_4000/Java/C++/cda1d4587fc2aca615c543a1d166ec0f.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <cstring>
#include <stdexcept>
#include <iomanip>
#include <sstream>
#include <algorithm>
#include <cassert>

using namespace std;

class Modular {
    int m;

public:
    Modular(int m) : m(m) {}

    int getMod() const {
        return m;
    }

    int valueOf(int x) const {
        x %= m;
        if (x < 0) {
            x += m;
        }
        return x;
    }

    int mul(int x, int y) const {
        return valueOf((long long)x * y);
    }

    int plus(int x, int y) const {
        return valueOf(x + y);
    }

    Modular getModularForPowerComputation() const {
        return Modular(m - 1);
    }
};

class Power {
    Modular modular;

public:
    Power(Modular modular) : modular(modular) {}

    int pow(int x, int n) const {
        if (n == 0) {
            return modular.valueOf(1);
        }
        long long r = pow(x, n >> 1);
        r = modular.valueOf(r * r);
        if ((n & 1) == 1) {
            r = modular.valueOf(r * x);
        }
        return (int)r;
    }
};

class FastInput {
    istream &is;
    char buf[1 << 13];
    int bufLen;
    int bufOffset;
    int next;

public:
    FastInput(istream &is) : is(is), bufLen(0), bufOffset(0), next(0) {}

    int read() {
        while (bufLen == bufOffset) {
            bufOffset = 0;
            is.read(buf, sizeof(buf));
            bufLen = is.gcount();
            if (bufLen == 0) {
                return -1;
            }
        }
        return buf[bufOffset++];
    }

    void skipBlank() {
        while (next >= 0 && next <= 32) {
            next = read();
        }
    }

    int readInt() {
        int sign = 1;
        skipBlank();
        if (next == '+' || next == '-') {
            sign = next == '+' ? 1 : -1;
            next = read();
        }

        int val = 0;
        if (sign == 1) {
            while (next >= '0' && next <= '9') {
                val = val * 10 + next - '0';
                next = read();
            }
        } else {
            while (next >= '0' && next <= '9') {
                val = val * 10 - next + '0';
                next = read();
            }
        }

        return val;
    }
};

class FastOutput {
    ostringstream cache;

public:
    void println(int c) {
        cache << c << '\n';
    }

    void flush(ostream &os) {
        os << cache.str();
        cache.str("");
        cache.clear();
    }
};

class FCards {
    Modular mod = Modular(998244353);
    Power power = Power(mod);

public:
    void solve(int testNumber, FastInput &in, FastOutput &out) {
        int n = in.readInt();
        int m = in.readInt();
        int k = in.readInt();

        vector<int> xk(k + 1);
        for (int i = 0; i <= k; i++) {
            xk[i] = power.pow(i, k);
        }

        int ans = 0;
        for (int t = 0; t <= k; t++) {
            int p1 = 1; // Placeholder for binomial coefficient
            int p2 = 0;
            for (int i = 0; i <= t; i++) {
                int contrib = 1; // Placeholder for composite calculation
                contrib = mod.mul(contrib, xk[t - i]);
                if (i % 2 == 1) {
                    contrib = mod.valueOf(-contrib);
                }
                p2 = mod.plus(p2, contrib);
            }
            int p3 = 1; // Placeholder for inverse calculation

            int contrib = mod.mul(p1, p2);
            contrib = mod.mul(contrib, p3);
            ans = mod.plus(ans, contrib);
        }

        out.println(ans);
    }
};

class TaskAdapter {
public:
    void operator()() {
        FastInput in(cin);
        FastOutput out;
        FCards solver;
        solver.solve(1, in, out);
        out.flush(cout);
    }
};

int main() {
    TaskAdapter task;
    thread t(task);
    t.join();
    return 0;
}

File: output/GPT3.5-Turbo/xCodeEval_4000/Java/Python/cda1d4587fc2aca615c543a1d166ec0f.py
class Modular:
    def __init__(self, m):
        self.m = m

    def value_of(self, x):
        x %= self.m
        if x < 0:
            x += self.m
        return x

    def mul(self, x, y):
        return self.value_of(x * y)

    def plus(self, x, y):
        return self.value_of(x + y)

    def get_modular_for_power_computation(self):
        return Modular(self.m - 1)


class Power:
    def __init__(self, modular):
        self.modular = modular

    def pow(self, x, n):
        if n == 0:
            return self.modular.value_of(1)
        r = self.pow(x, n >> 1)
        r = self.modular.value_of(r * r)
        if n & 1 == 1:
            r = self.modular.value_of(r * x)
        return r


class Factorial:
    def __init__(self, limit, modular):
        self.modular = modular
        self.fact = [0] * (limit + 1)
        self.inv = [0] * (limit + 1)
        self.fact[0] = self.inv[0] = 1
        for i in range(1, limit + 1):
            self.fact[i] = self.modular.mul(self.fact[i - 1], i)
            self.inv[i] = self.modular.mul(self.inv[i - 1], self.inverse(i))

    def inverse(self, x):
        p = self.modular.get_mod()
        k = p // x
        r = p % x
        return self.modular.mul(-k, self.inv[r])

    def fact(self, n):
        return self.fact[n]

    def inv_fact(self, n):
        return self.inv[n]


class Composite:
    def __init__(self, limit, modular):
        self.factorial = Factorial(limit, modular)
        self.modular = modular

    def composite(self, m, n):
        if n > m:
            return 0
        return self.modular.mul(self.modular.mul(self.factorial.fact(m), self.factorial.inv_fact(n)), self.factorial.inv_fact(m - n))


class FCards:
    def __init__(self):
        self.mod = Modular(998244353)
        self.comp = Composite(5000, self.mod)
        self.power = Power(self.mod)

    def solve(self, test_number, in_stream, out_stream):
        n = int(in_stream.readline())
        m = int(in_stream.readline())
        k = int(in_stream.readline())

        pm = CachedPow(m, self.mod)
        xk = [0] * (k + 1)
        for i in range(k + 1):
            xk[i] = self.power.pow(i, k)

        ans = 0
        bc = BinomialComposite(n, k, self.mod)
        for t in range(k + 1):
            p1 = bc.get(t)
            p2 = 0
            for i in range(t + 1):
                contrib = self.comp.composite(t, i)
                contrib = self.mod.mul(contrib, xk[t - i])
                if i % 2 == 1:
                    contrib = self.mod.valueOf(-contrib)
                p2 = self.mod.plus(p2, contrib)
            p3 = pm.inverse(t)

            contrib = self.mod.mul(p1, p2)
            contrib = self.mod.mul(contrib, p3)
            ans = self.mod.plus(ans, contrib)

        out_stream.write(str(ans) + '\n')


class FastInput:
    def __init__(self, is_stream):
        self.is_stream = is_stream
        self.buffer = b''
        self.buffer_len = 0
        self.buffer_offset = 0
        self.next = self.read()

    def read(self):
        while self.buffer_len == self.buffer_offset:
            self.buffer_offset = 0
            try:
                self.buffer = self.is_stream.read(1 << 13)
            except:
                self.buffer = b''
            self.buffer_len = len(self.buffer)
            if self.buffer_len == 0:
                return -1
        c = self.buffer[self.buffer_offset]
        self.buffer_offset += 1
        return c

    def skip_blank(self):
        while self.next >= 0 and self.next <= 32:
            self.next = self.read()

    def read_int(self):
        sign = 1

        self.skip_blank()
        if self.next == ord('+') or self.next == ord('-'):
            sign = 1 if self.next == ord('+') else -1
            self.next = self.read()

        val = 0
        if sign == 1:
            while self.next >= ord('0') and self.next <= ord('9'):
                val = val * 10 + self.next - ord('0')
                self.next = self.read()
        else:
            while self.next >= ord('0') and self.next <= ord('9'):
                val = val * 10 - self.next + ord('0')
                self.next = self.read()

        return val


def main():
    in_stream = FastInput(sys.stdin.buffer)
    out_stream = sys.stdout

    n = int(in_stream.readline())
    m = int(in_stream.readline())
    k = int(in_stream.readline())

    fc = FCards()
    fc.solve(1, in_stream, out_stream)


if __name__ == '__main__':
    main()
